conn/physic/units_test.go

```diff
@@ -536,3 +538,269 @@ func BenchmarkCelsiusFloatg(b *testing.B) {
 		buf.Reset()
 	}
 }
+
+func TestAtod(t *testing.T) {
+	const (
+		negative = true
+		positive = false
+	)
+	succeeds := []struct {
+		in       string
+		expected decimal
+		n        int
+	}{
+		{"123456789", decimal{"123456789", 0, positive}, 9},
+		{"1nM", decimal{"1", 0, positive}, 1},
+		{"2.2nM", decimal{"22", -1, positive}, 3},
+		{"12.5mA", decimal{"125", -1, positive}, 4},
+		{"-12.5mA", decimal{"125", -1, negative}, 5},
+		{"1ma1", decimal{"1", 0, positive}, 1},
+		{"+1ma1", decimal{"1", 0, positive}, 2},
+		{"-1ma1", decimal{"1", 0, negative}, 2},
+		{"-0.00001%rH", decimal{"1", -5, negative}, 8},
+		{"0.00001%rH", decimal{"1", -5, positive}, 7},
+		{"1.0", decimal{"1", 0, positive}, 3},
+		{"0.10001", decimal{"10001", -5, positive}, 7},
+		{"+0.10001", decimal{"10001", -5, positive}, 8},
+		{"-0.10001", decimal{"10001", -5, negative}, 8},
+		{"1n", decimal{"1", 0, positive}, 1},
+		{"1.n", decimal{"1", 0, positive}, 2},
+		{"-1.n", decimal{"1", 0, negative}, 3},
+		{"200n", decimal{"2", 2, positive}, 3},
+		{".01", decimal{"1", -2, positive}, 3},
+		{"+.01", decimal{"1", -2, positive}, 4},
+		{"-.01", decimal{"1", -2, negative}, 4},
+		{"1-2", decimal{"1", 0, positive}, 1},
+		{"1+2", decimal{"1", 0, positive}, 1},
+		{"-1-2", decimal{"1", 0, negative}, 2},
+		{"-1+2", decimal{"1", 0, negative}, 2},
+		{"+1-2", decimal{"1", 0, positive}, 2},
+		{"+1+2", decimal{"1", 0, positive}, 2},
+		{"010", decimal{"1", 1, positive}, 3},
+		{"001", decimal{"1", 0, positive}, 3},
+	}
+
+	fails := []struct {
+		in       string
+		expected decimal
+		n        int
+	}{
+		{"1.1.1", decimal{}, 0},
```

Comment:
I'm not ambivalent that it should fail; it could succeed and return "1.1" and ".1" as the trailing string.
But at this point let's commit since this code is already superb.